src/fmu/tools/qcproperties/qcproperties.py

```diff
@@ -0,0 +1,242 @@
+"""The qcproperties module"""
+from pathlib import Path
+
+import pandas as pd
+import yaml
+
+from fmu.tools.qcforward._qcforward_data import _QCForwardData
+
+from fmu.tools.qcproperties._combine_propstats import combine_property_statistics
+from fmu.tools.qcproperties._propstat_parameter_data import PropStatParameterData
+from fmu.tools.qcproperties._propstat import PropStat
+
+
+class QCProperties:
+    """
+    The QCProperties class consists of a set of methods for extracting
+    property statistics from 3D Grids, Raw and Blocked wells.
+
+    The methods for statistics extraction can be run individually, or a
+    yaml-configuration file can be used to enable an automatic run of the
+    methods. See the method 'from_yaml'.
+
+    When several methods of statistics extraction has been run within the instance,
+    a merged dataframe is available through the 'dataframe' property.
+
+    All methods can be run from either RMS python, or from files.
+
+    XTGeo is being utilized to get a dataframe from the input parameter data.
+    XTGeo data is reused in the instance to increase performance.
+
+
+    Methods for extracting statistics from 3D Grids, Raw and Blocked wells:
+
+        Args:
+            data (dict): The input data as a Python dictionary (see description of
+                valid argument keys in documentation)
+            reuse (bool or list): If True, then grid and gridprops will be reused
+                as default. Alternatively it can be a list for more
+                fine grained control, e.g. ["grid", "gridprops", "wells"]
+            project (obj or str): For usage inside RMS
+
+        Returns:
+            A PropStat() instance
+
+    """
+
+    def __init__(self):
+        self._gdata = None  # QCForwardData instance, general data
```

Comment:
The `QCProperties` class should be independent of the `QCForward` class, but currently the data loading is done using `QCForwardData`.

Perhaps we should consider moving the data loading out of the `QCForward` code base?

Thoughts @jcrivenaes ?